GitUI/CommandsDialogs/SettingsDialog/Pages/ShellExtensionSettingsPage.cs

```diff
@@ -46,6 +66,25 @@ protected override void PageToSettings()
 
             AppSettings.CascadeShellMenuItems = l_CascadeShellMenuItems;
             AppSettings.AlwaysShowAllCommands = cbAlwaysShowAllCommands.Checked;
+
+            if (ShellExtensionManager.CheckFilesFound() && cbEnableIntegration.Checked != ShellExtensionManager.CheckIfRegistered())
+            {
+                try
+                {
+                    if (cbEnableIntegration.Checked)
+                    {
+                        ShellExtensionManager.Register();
+                    }
+                    else
+                    {
+                        ShellExtensionManager.Unregister();
+                    }
+                }
+                catch
+                {
+                    // ignored
+                }
```

Comment:
Why? Either `ShellExtensionManager` throws or it has handled the exceptions.

Reply:
1. `ShellExtensionManager` can throw exceptions.
2. In this case the exceptions are not critical so I have to catch them.
3. Moreover as far as I get it in PageToSettings I cannot even show any errors so I can only ignore errors from `ShellExtensionManager`.